app/src/main/java/org/simple/clinic/editpatient/EditPatientScreen.kt

```diff
@@ -410,7 +410,13 @@ class EditPatientScreen : BaseScreen<
   }
 
   override fun openSimpleScanIdScreen(openedFrom: OpenedFrom) {
-    router.pushExpectingResult(ScanBpPassport, ScanSimpleIdScreenKey(openedFrom))
+    val requestType = when (openedFrom) {
+      is OpenedFrom.EditPatientScreen.ToAddBpPassport -> ScanBpPassport
+      is OpenedFrom.EditPatientScreen.ToAddNHID -> ScanIndiaNationalHealthID
+      else -> throw IllegalArgumentException("Opened from parameter is unknown, this shouldn't happen")
```

Comment:
I chose this earlier because we were not opening EditPatientScreen from anything here. So sounded incorrect. but we could be referring to the parameter itself also.